### PR TITLE
Add a GitLab CI config example

### PR DESCRIPTION
**Pull request by LeoColomb**
Thanks to the core reorganization, the `.gitlab-ci.yml` can be very generic (thanks! 😃).

This allow to get a working dependabot right* by mirror this repo. 🎉 

<small>* Of course a local configuration (basic doc included in the file) and an available gitlab-runner are required.</small>

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+# https://docs.gitlab.com/ee/ci/yaml/
+
+.dependabot:
+  image: dependabot/dependabot-core
+  variables:
+    BUNDLE_SILENCE_ROOT_WARNING: 1
+    PACKAGE_MANAGER: $CI_JOB_NAME
+  before_script:
+    - bundle install -j $(nproc) --path vendor
+    - bundle exec irb
```
[the remark this answers is not in the document]
Comment (LeoColomb) [reply .gitlab-ci.yml]: I don't know at all, this is just what the [README](https://github.com/dependabot/dependabot-script/blob/938757071ce850a8d4f736f2ca14bc32a09e7fd0/README.md#setup-and-usage) explains.
Maybe not? Let me know.

**Comment by greysteil** [comment]
This looks awesome. I wonder about renaming the file to end with `.example`, though, and adding a block comment to the top of it explaining that you need to copy that file and the updater script to make things run?

File: .gitlab-ci.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (LeoColomb) [reply .gitlab-ci.yml]: Oh, is this for interactive ruby?

[the remark this answers is not in the document]
Comment (greysteil) [reply .gitlab-ci.yml]: Ah, no, you can kill this line. It opens an interactive ruby sessions, which is where you'd normally paste the commands you want to run. When you're running a script that's not necessary 🙂

**Comment by LeoColomb** [comment]
@greysteil I can surely append a `.example`, but this will remove the ability to just mirror this repo to enable runs.

**Comment by greysteil** [comment]
Ahh, I've just realised how you're intending folks to use this - you're expecting them to mirror this repo on GitLab and then set it to do the dependency check on a cron from that repo, targeting another repo (that will receive the updates), right?

In that case, it would be awesome to have an explanation of that in the README. I think I'd still favour calling the file `.gitlab-ci.yml.example`, just because I wouldn't want people to get any surprises when they initially set up the mirror (and since we're not making any guarantees about not making breaking changes to the way the script works I wouldn't want them blindly updating it and assuming nothing will have changed).

**Comment by LeoColomb** [comment]
> do the dependency check on a cron from that repo, targeting another repo

Exactly! 😇 

> when they initially set up the mirror

Oh, good point, you're right.
May I suggest `.gitlab-ci.example.yml` instead? (basically for syntax highlighting)

**Comment by greysteil** [comment]
Of course, 👍 on that!

**Comment by LeoColomb** [comment]
Done, with documentation! 🎉

**Comment by greysteil** [comment]
Awesome!

**Pull request merged**